table.cc

```diff
@@ -2344,3 +2355,55 @@ table::as_mutation_source_excluding(std::vector<sstables::shared_sstable>& ssts)
         return this->make_reader_excluding_sstables(std::move(s), std::move(permit), ssts, range, slice, pc, std::move(trace_state), fwd, fwd_mr);
     });
 }
+
+future<> table::update_repair_time(dht::token_range range, gc_clock::time_point repair_time) {
+    _repair_time_map->map += std::make_pair(locator::token_metadata::range_to_interval(range), repair_time);
+    return make_ready_future<>();
+}
+
+gc_clock::time_point
+table::get_gc_before(const dht::decorated_key& dk, const gc_clock::time_point& query_time) const {
+    // if mode = timeout    // default option, if user does not specify tombstone_gc options
+    // if mode = disabled   // never gc tombstone
+    // if mode = immediate  // can gc tombstone immediately
+    // if mode = repair     // gc after repair
+    const auto& options = _schema->tombstone_gc_options();
+    switch (options.mode()) {
+    case tombstone_gc_mode::timeout:
+        tlogger.info("Get gc_beofre for ks={}, table={}, dk={}, mode=timeout", schema()->ks_name(), schema()->cf_name(), dk);
+        return saturating_subtract(query_time, _schema->gc_grace_seconds());
```

Comment:
kudos for the name timeout. feels like we need to make this log debug level? 

s/gc_beofre/gc_before

Reply:
It is supposed to be trace. I did not bring it back during development for debug. 

Typo. Fixed. Thanks.